ase/Makefile

```diff
@@ -243,6 +226,55 @@ ASE_LD_SWITCHES+= -lrt -lpthread
 ASE_SHOBJ_NAME = ase_libs
 ASE_SHOBJ_SO = $(ASE_SHOBJ_NAME).so
 
+#########################################################################
+#                 Enable Altera gates library in ASE                    #
+#########################################################################
+# Enable Altera Gates
+GLS_SIM = 1
+
+# What is the target FPGA family? The definition comes from the PIM
+# in rtl/ase_platform_config.mk.
+FPGA_FAMILY ?= UNKNOWN
+ifdef PLATFORM_FPGA_FAMILY_A10
+  FPGA_FAMILY = arria10
+endif
+ifdef PLATFORM_FPGA_FAMILY_S10
+  FPGA_FAMILY = stratix10
+endif
+$(info FPGA_FAMILY=$(FPGA_FAMILY))
+$(info )
+
+# Gate level libraries to add to simulation
+GLS_VERILOG_OPT ?=
+QSIM_MODEL_VLOG_OPT ?=
+
+# The Quartus library simulation targets change based on the simulator.
```

Comment:
Will these Quartus library simulation targets be used by ASE?

Reply:
Yes, by the user RTL and Qsys-generated code.

Reply:
Will these only be used some user RTL which needs to generate RTL files from Qsys? Thank you!

Reply:
It is possible to write RTL that doesn't need them, though most real programs will need Quartus libraries. There's no way to probe the code and figure it out, so we now use Quartus to tell ASE how to compile the base libraries. It only takes a minute or two and it is only done once, even if make is invoked multiple times.